app/code/Magento/Eav/Model/Entity/Attribute/Source/AbstractSource.php

```diff
@@ -74,7 +74,7 @@ public function getOptionText($value)
             }
         }
         // End
-        if (is_scalar($value) && isset($options[$value])) {
+        if (is_scalar($value) && isset($options[$value]) && !is_array($options[$value])) {
```

Comment:
Hi @woutersamaey ,
Seems like we still might have a case when not declared type could be returned, for instance if the value is integer, float, resource, object, etc.

I would suggest to explicitly check if it's a string, as bool value should be returner only in case if we weren't able to find a text.

```suggestion
        if (is_scalar($value) && isset($options[$value]) && is_string($options[$value])) {
```

What do you thing?